evalml/pipelines/components/transformers/imputers/imputer.py

```diff
@@ -134,6 +131,5 @@ def transform(self, X, y=None):
             X_categorical = X.ww[self._categorical_cols.tolist()]
             imputed = self._categorical_imputer.transform(X_categorical)
             X_no_all_null[X_categorical.columns] = imputed
```

Comment:
Does `imputed` have a valid ww schema because one is initialized in the `transform` call? Or is it the same object as `X_categorical`?

Either way, as long as `imputed` doesn't change the dtype (and having the same logical types would mean the dtypes were the same), it wouldn't invalidate the schema when it gets added to `X_no_all_null` (this is assuming you guys still aren't using a woodwork index).

But if the columns in `X_categorical` are not a subset of those in `X_no_all_null` that could also invalidate the schema (maybe if a categorical column was somehow fully null?).  

This operation can happen outside of Woodwork and not invalidate the schema, but it's more of a coincidence than any thing else. So if there's ever a chance the woodwork schema could be invalidated in either of the ways described above, you'd want a way of going through woodwork. One option would be to get the bulk column update with a pandas method that woodwork hasn't overwritten--like `X_no_all_null.ww.update(imputed)` or something

Reply:
Interestingly enough, `X_no_all_null.ww.update(imputed)` wipes out the woodwork initialization for `X_no_all_null`. Re-initializing `X_no_all_null` afterward works, but I'm not sure if it's any better than what we currently have. Thoughts?

Reply:
That means that the update call invalidated the schema. Can you see the `TypingInfoMismatchWarning` message that's being produced?

The fact that the schema is invalidated is either a side effect of pandas' `update`, or it's an indication that the resulting `X_no_all_null ` dataframe never had a valid schema but that wasn't caught _because_ none of the changes happened inside of woodwork. If you revert back to the way it was here (so not going through woodwork) but add an assertion that `ww.is_schema_valid(X_no_all_null, X_no_all_null.ww.schema)` and the assertion passes, then it's likely the `update` call. Otherwise, it's that the schema was never valid, so going through woodwork will be the route to take.

Reply:
The full warning is this: 
```
evalml/tests/component_tests/test_imputer.py::test_imputer_does_not_erase_ww_info
  Operation performed by update has invalidated the Woodwork typing information:
   dtype mismatch for column b between DataFrame dtype, object, and Categorical dtype, category.
   Please initialize Woodwork with DataFrame.ww.init
```
When I set the assertion of `ww.is_schema_valid`, a single test failed. However, that test mocked the `SimpleImputer.transform`, in which case `imputed` was not a woodwork object (which it's otherwise guaranteed to be).

Reply:
okay, that sounds to me like update is just changing the dtypes to `object`. So probably not worth going through woodwork for the update all. I'll leave it up to you guys whether you're comfortable with assuming the schema is valid after these changes happen outside of woodwork or whether you want to formally call `init_with_full_schema` as a sanity check that that's actually true

Reply:
@tamargrey Thank you for your thoughts on this! I think for now we can assume that the dtypes will always match. That's because the features without missing values will not change and the features with missing values can only be Double or Categorical in which case the imputation will not change the dtype. Of course, that's only because nullable types are currently not allowed.